gtsam/basis/Chebyshev2.h

```diff
@@ -0,0 +1,137 @@
+/* ----------------------------------------------------------------------------
+
+ * GTSAM Copyright 2010, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * Authors: Frank Dellaert, et al. (see THANKS for the full author list)
+
+ * See LICENSE for the license information
+
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file Chebyshev2.h
+ * @brief Chebyshev parameterizations on Chebyshev points of second kind.
+ *        This is different from Chebyshev.h since it leverage ideas from
+ *        pseudo-spectral optimization, i.e. we don't decompose into basis
+ *        functions, rather estimate function parameters that enforce function
+ *        nodes at Chebyshev points.
+ * @author Varun Agrawal, Frank Dellaert
+ * @date July 4, 2020
+ */
+
+#pragma once
+
+#include <gtsam/base/Manifold.h>
+#include <gtsam/base/OptionalJacobian.h>
+#include <gtsam/basis/Basis.h>
+
+#include <boost/function.hpp>
+
+namespace gtsam {
+
+/**
+ * Chebyshev Interpolation on Chebyshev points of the second kind
+ * Note that N here, the #points, is one less than N from Trefethen00book
+ * (pg.42)
+ */
```

Comment:
Maybe add a reference here to book

Reply:
Isn't `Trefenthen00Book` the reference in that comment?